link_test.go

```diff
@@ -664,3 +667,151 @@ func TestLinkStats64UnmarshalBinary(t *testing.T) {
 		})
 	}
 }
+
+func TestLinkXDP(t *testing.T) {
+	// BPF compilation requires a high RLIMIT_MEMLOCK
```

Comment:
Compilation doesn't require this, technically we're loading byte-code so there is no compilation happening, we need this for _loading_ eBPF programs and it only applies to this process so it's not dirtying the rest of the system.